pkg/network/state.go

```diff
@@ -507,10 +507,10 @@ func (ns *networkState) storeDNSStats(stats dns.StatsByKeyByNameByType) {
 	// Fast-path for common case (one client registered)
 	if len(ns.clients) == 1 {
 		for _, c := range ns.clients {
-			if len(c.dnsStats) == 0 {
+			if len(c.dnsStats) == 0 && getDeepDNSStatsCount(stats) <= ns.maxDNSStats {
```

Comment:
is there any way to enforce that `stats` does not grow beyond the limit in the first place?

Reply:
There is a separate limit elsewhere for that it looks like: https://github.com/datadog/datadog-agent/blob/hasan.mahmood/fix-dns-stats-limit/pkg/network/dns/snooper.go#L82

Reply:
I know this PR doesn't really add any new limit but merely fixes the enforcement of an existing one - but I am curious to know why we need both for DNS stats. If there is an obvious distinction that I am missing, addition of a comment may be helpful.

Reply:
I think the limit in state.go is meant as an upper limit on how many stats we report in the connections check. If the other limit in snooper was > 75000 (the hardcoded state.go limit), we would drop data in state.go.